lib/ansible/modules/cloud/amazon/iam_group.py

```diff
@@ -282,25 +288,33 @@ def create_or_update_group(connection, module):
 
         if purge_users:
             for user in list(set(current_group_members_list) - set(users)):
-                try:
-                    connection.remove_user_from_group(GroupName=params['GroupName'], UserName=user)
-                except ClientError as e:
-                    module.fail_json(msg=e.message, exception=traceback.format_exc(),
-                                     **camel_dict_to_snake_dict(e.response))
-                except ParamValidationError as e:
-                    module.fail_json(msg=e.message, exception=traceback.format_exc())
-
-        if users != [None]:
-            for user in users:
-                try:
-                    connection.add_user_to_group(GroupName=params['GroupName'], UserName=user)
-                except ClientError as e:
-                    module.fail_json(msg=e.message, exception=traceback.format_exc(),
-                                     **camel_dict_to_snake_dict(e.response))
-                except ParamValidationError as e:
-                    module.fail_json(msg=e.message, exception=traceback.format_exc())
-
-        changed = True
+                # Ensure we mark things have changed if any user gets purged
+                changed = True
+                # Skip actions for check mode
+                if not module.check_mode:
+                    try:
+                        connection.remove_user_from_group(GroupName=params['GroupName'], UserName=user)
+                    except ClientError as e:
+                        module.fail_json(msg=e.message, exception=traceback.format_exc(),
+                                         **camel_dict_to_snake_dict(e.response))
+                    except ParamValidationError as e:
+                        module.fail_json(msg=e.message, exception=traceback.format_exc())
+        # If there are users to adjust that aren't in the current list, then things have changed
+        # Otherwise the only changes were in purging above
+        if set(users).difference(set(current_group_members_list)):
```

Comment:
Yeah, sorry, you're right, that's fairly obvious now you point that out. Again, we can use standard set operators for this

```
if set(users) - set(current_group_members_list):
```